### PR TITLE
update OpenSSL to v1.1.1i

### PR DESCRIPTION
**Pull request by niall-robson-instil**

### CODE REVIEW

**Comment by Paul-Shields** [comment]
So the build scripts here work unmodified with this version of OpenSSL?

File: toolchains_path.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python
```
Comment (Paul-Shields) [inline toolchains_path.py]: Can we remove this file?

File: build-android.sh [hunk 2]
```diff
@@ -1,159 +1,92 @@
...
         exit 1
         ;;
 esac
 
-rm -rf ${ANDROID_LIB_ROOT} > /dev/null
-rm -rf ${TMP_DIR} > /dev/null
-mkdir ${TMP_DIR} > /dev/null
+ANDROID_API=21
+CC=clang
+echo $toolchains_path/bin
```
Comment (Paul-Shields) [inline build-android.sh]: Is this just a debug line?

```suggestion
```

Comment (Paul-Shields) [reply build-android.sh]: (These GitHub PR tools are really good!)

File: build-android.sh [hunk 3]
```diff
@@ -1,159 +1,92 @@
...
-        else
-            TOOLCHAIN_SYSTEM=linux-x86_64
-        fi
+        toolchains_path=${ANDROID_NDK_HOME}/toolchains/llvm/prebuilt/linux-x86_64
         ;;
     *)
         echo "Toolchain unknown for host system"
         exit 1
         ;;
 esac
```
Comment (Paul-Shields) [inline build-android.sh]: I know this is just tweaking what was there before, but I'm guessing this won't work on the M1 mac. Happy to test that out, but I think with some alternate uses of `uname` we can simplify this a bit

```suggestion
toolchains_path=/toolchains/llvm/prebuilt/$(uname -s | tr A-Z a-z)-$(uname -m)
```

Or maybe 

```suggestion
OS_TYPE=$(uname -s | tr A-Z a-z)
ARCH=$(uname -m)
toolchains_path=/toolchains/llvm/prebuilt/$OS_TYPE-$ARCH
```

**Review by Paul-Shields — CHANGES_REQUESTED** [review]

File: toolchains_path.py (hunk 1 above)
Comment (niall-robson-instil) [reply toolchains_path.py]: yea

File: build-android.sh (hunk 3 above)
Comment (Paul-Shields) [reply build-android.sh]: Sorry, sugggestion is slightly wrong, forgot the `${ANDROID_NDK_HOME}` from the `toolchains_path`. Got too excited with the fancy tools 😄

**Pull request merged**